packages/utils/src/timeout.ts

```diff
@@ -1,12 +1,16 @@
 import {anySignal} from "any-signal";
-import {TimeoutError} from "./errors.js";
+import {ErrorAborted, TimeoutError} from "./errors.js";
 import {sleep} from "./sleep.js";
 
 export async function withTimeout<T>(
   asyncFn: (timeoutAndParentSignal?: AbortSignal) => Promise<T>,
   timeoutMs: number,
   signal?: AbortSignal
 ): Promise<T> {
+  if (signal?.aborted) {
```

Comment:
this seems nice-to-have as we don't really wanna execute the `asyncFn` if parent signal is aborted